src/NuGet.Core/NuGet.Common/Telemetry/TelemetryServiceUtility.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace NuGet.Common
+{
+    public static class TelemetryServiceUtility
+    {
+        private static Stopwatch _stopWatch;
+
+        public static void StartorResumeTimer()
+        {
+            if (_stopWatch == null)
+            {
+                _stopWatch = Stopwatch.StartNew();
+            }
+            else
+            {
+                _stopWatch.Start();
+            }
+        }
+
+        public static void StopTimer()
+        {
+            _stopWatch?.Stop();
+        }
+
+        public static TimeSpan GetTimerElapsedTime()
+        {
+            var duration = new TimeSpan();
+
+            if (_stopWatch != null)
+            {
+                duration = _stopWatch.Elapsed;
+                _stopWatch.Reset();
+            }
+
+            return duration;
```

Comment:
`return duration ?? TimeSpan.MinValue;`

That will avoid creating a new TimeSpan each time. Set duration to null above, or just return it in the if and return MinValue outside of the if.